src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -146,129 +122,123 @@ class PopoverReportActionContextMenu extends React.Component {
         isChronosReport = false,
         isPinnedChat = false,
         isUnreadChat = false,
-    ) {
+    ) => {
         const nativeEvent = event.nativeEvent || {};
-        this.contextMenuAnchor = contextMenuAnchor;
-        this.contextMenuTargetNode = nativeEvent.target;
-
-        // Singleton behaviour of ContextMenu creates race conditions when user requests multiple contextMenus.
-        // But it is possible that every new request registers new callbacks thus instanceID is used to corelate those callbacks
-        this.instanceID = Math.random().toString(36).substr(2, 5);
-
-        this.onPopoverShow = onShow;
-        this.onPopoverHide = onHide;
-
-        this.getContextMenuMeasuredLocation().then(({x, y}) => {
-            this.setState({
-                cursorRelativePosition: {
-                    horizontal: nativeEvent.pageX - x,
-                    vertical: nativeEvent.pageY - y,
-                },
-                popoverAnchorPosition: {
-                    horizontal: nativeEvent.pageX,
-                    vertical: nativeEvent.pageY,
-                },
-                type,
-                reportID,
-                reportAction,
-                selection,
-                isPopoverVisible: true,
-                reportActionDraftMessage: draftMessage,
-                isArchivedRoom,
-                isChronosReport,
-                isPinnedChat,
-                isUnreadChat,
-            });
+        contextMenuAnchor.current = contextMenuAnchor;
+        contextMenuTargetNode.current = nativeEvent.target;
+
+        setInstanceID(Math.random().toString(36).substr(2, 5));
+
+        onPopoverShow.current = onShow;
+        onPopoverHide.current = onHide;
+
+        getContextMenuMeasuredLocation().then(({x, y}) => {
+            popoverAnchorPosition.current = {
+                horizontal: nativeEvent.pageX - x,
+                vertical: nativeEvent.pageY - y,
+            };
+
+            popoverAnchorPosition.current = {
+                horizontal: nativeEvent.pageX,
+                vertical: nativeEvent.pageY,
+            };
+            typeRef.current = type;
+            reportIDRef.current = reportID;
+            reportActionRef.current = reportAction;
+            selectionRef.current = selection;
+            setIsPopoverVisible(true);
+            reportActionDraftMessageRef.current = draftMessage;
+            setIsArchivedRoom(isArchivedRoom);
+            setIsChronosReport(isChronosReport);
+            setIsPinnedChat(isPinnedChat);
+            setIsUnreadChat(isUnreadChat);
         });
-    }
+    };
 
     /**
      * This gets called on Dimensions change to find the anchor coordinates for the action context menu.
      */
-    measureContextMenuAnchorPosition() {
-        if (!this.state.isPopoverVisible) {
+    const measureContextMenuAnchorPosition = useCallback(() => {
```

Comment:
this function I think should be above the `useEffect` that uses it, also because that `useEffect` doesn't depend on anything I think you don't need to wrap this `measureContextMenuAnchorPosition` into a `useCallback`

Reply:
if you wanna keep it in a `useCallback` this function `getContextMenuMeasuredLocation` must be in a `useCallback` too

Reply:
Actually `useEffect` should have a dependency, my bad. So I kept `useCallback` here and added it to `getContextMenuMeasuredLocation.` Also moved these functions above `useEffect.`